src/librustc/dep_graph/graph.rs

```diff
@@ -70,6 +75,15 @@ impl DepGraph {
         raii::DepTask::new(&self.data.thread, key)
     }
 
+    /// Debugging aid -- forbid reads/writes to `key` while the return
+    /// value is in scope. Note that this is only available when debug
+    /// assertions are enabled -- you should not check in code that
+    /// invokes this function.
+    #[cfg(debug_assertions)]
```

Comment:
Nice. Ideally, I'd like to have an extended version of this in the future that's not just for ad hoc debugging.